lib/libutee/include/rng_pta.h

```diff
@@ -0,0 +1,43 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (C) 2021, Foundries.io Ltd
+ */
+
+#ifndef __RNG_PTA__H
+#define __RNG_PTA__H
+
+#define PTA_RNG_UUID { 0x035a4479, 0xc369, 0x47f4, \
+		{ 0x94, 0x51, 0xc6, 0xfd, 0xff, 0x28, 0xad, 0x65 } }
+
+/*
+ * PTA_CMD_GET_ENTROPY - Get Entropy from RNG using crypto_rng_read()
+ *
+ * param[0] (inout memref) - Entropy buffer memory reference
+ * param[1] unused
+ * param[2] unused
+ * param[3] unused
+ *
+ * Result:
+ * TEE_SUCCESS - Invoke command success
+ * TEE_ERROR_BAD_PARAMETERS - Incorrect input param
+ * TEE_ERROR_NOT_SUPPORTED - Requested entropy size greater than size of pool
+ * TEE_ERROR_HEALTH_TEST_FAIL - Continuous health testing failed
+ */
+#define PTA_CMD_GET_ENTROPY		0x0
+
+/*
+ * PTA_CMD_GET_RNG_INFO - Get RNG information
+ *
+ * param[0] (out value) - value.a: RNG data-rate in bytes per second
+ *                        value.b: Quality/Entropy per 1024 bit of data
+ * param[1] unused
+ * param[2] unused
+ * param[3] unused
```

Comment:
That is fine for now, but I'm wondering how to make this easily extensible in case we want to add something later (RNG name or whatever). How about adding `[out] value[1].a: zero`? Then we can later use this field as a version number.

Reply:
sure, sounds good to me

Reply:
the problem with that is that we'll have to update the REE side: in the case of Linux for instance this will mean modifying the current upstream driver (drivers/char/hw_random/optee-rng.c.)

maybe we can leave this change for a future update?

Reply:
@jforissier please let me know if I can add your Reviewed-by without this change

Reply:
I don't see the problem. If a new feature is needed by the REE, it has to be modified anyways. In this case we would add e.g., `if (param[1].value >= 1) { /* feature xyz supported */ }`. In order to be able to do that in the future the field has to be guaranteed to be zero in the first version.

Reply:
> In order to be able to do that in the future the field has to be guaranteed to be zero in the first version.

The first version is already [here](https://github.com/OP-TEE/optee_os/blob/master/core/arch/arm/plat-synquacer/rng_pta_client.h) for plat-synquacer. This pseudo TA is trying to leverage the existing optee-rng interface.

Reply:
@b49020 oh sorry I missed that. I don't really agree with the above discussion, and duplicating the PTAs. What's wrong with moving the existing `plat-synquacer` PTA into common code and extending it as needed? New commands can easily be added in a backward-compatible way.

Reply:
@jforissier IIUC, you meant to say that we should have a single RNG pTA that should provide weak default implementations for `rng_get_entropy()` and `rng_get_info()` and allow the platforms to override them, correct? If yes then I am fine with this approach as well.

Reply:
hi @b49020 so if this something you could do as part of the plat-synquacer maintainership? shall I just close this PR then?

Reply:
@b49020 yes I think it would work pretty well unless I missed something (quite possible TBH ;-))

Reply:
@ldts I am fine with you to continue this work. I think with platform override for `rng_get_info()`, we don't have to worry about RNG quality (as per your below comment) and we can't simply get away with configuring default implementation with PRNG quality.

Reply:
@b49020 on the contrary, please go ahead and do the changes if time allows (it is really your driver after all and this is probably low priority maintenance than development really...just adding a couple of hooks) . I'll close this PR